replicators/src/mysql_connector/snapshot.rs

```diff
@@ -665,6 +680,41 @@ impl MySqlReplicator {
 
         Ok(())
     }
+
+    /// This function drops all known tables from Noria. This is used when we require a full
+    /// re-snapshot of the database.
+    ///
+    /// # Arguments
+    /// * `noria`: The target ReadySet deployment
+    async fn drop_all_tables(
+        &mut self,
+        noria: &mut readyset_client::ReadySetHandle,
+    ) -> ReadySetResult<()> {
+        let replication_offsets = noria.replication_offsets().await?;
+        let mut tx = self.pool.start_transaction(tx_opts()).await?;
+        let all_tables = get_table_list(&mut tx, TableKind::BaseTable).await?;
+        let mut changes = Vec::new();
+        for (db, table) in all_tables.iter() {
+            let table = Relation {
+                schema: Some(db.clone().into()),
+                name: table.clone().into(),
+            };
+            if replication_offsets.has_table(&table) {
+                changes.push(Change::Drop {
+                    name: table,
+                    if_exists: true,
+                });
+            }
+        }
+        tx.rollback().await?;
```

Comment:
we can probably do this before building up the change list, yeah?

Reply:
True, I was not sure, as some drivers do not transfer the whole resultset. They advance it as you iterate through. 
But by looking closely into get_table_list, we get a vector with the changes. I will move it up.

Reply:
After the latest comment from @ethan-readyset , I have moved the drop function to inside `load_recipe_with_meta_lock` and used the `table_list` gathered from there. Now `drop_all_tables` don't interact with the upstream database anymore.